### DISCUSSION TITLE
bb with specter

### DISCUSSION THREAD

**Discussion by drorbemet** [babashka/babashka#1150] [Q&A]
Hi,
first, thanks for developing and sharing babashka! :-)

Now, 
after adding specter and it's dep riddley to the classpath,
requiring specter does result in:

Unable to resolve classname: clojure.lang.Var

``` clojure
#!/usr/bin/env bb

(babashka.classpath/add-classpath "./.m2/repository/riddley/riddley/0.2.0/riddley-0.2.0.jar")
(babashka.classpath/add-classpath "./.m2/repository/com/rpl/specter/1.1.3/specter-1.1.3.jar")

(println (babashka.classpath/get-classpath))

(ns user
  (:require [com.rpl.specter :as spr
             :refer [ALL MAP-VALS select transform]]))

```

I looked at 
https://github.com/babashka/sci/issues/370
https://gist.github.com/borkdude/f6abcb0ff4183df559345503cb9c621c
and
https://github.com/borkdude/specter-cli/blob/main/src/specter_sci.clj

But now I ask myself what are the ways to get the adapted version of specter into a babashka repl at the moment.

I didn't find a babashka.specter namespace like e.g. for raynes.fs babashka.fs.

Would recompiling babashka combined with an adapted specter source solve the issue?

Is there a way to avoid this fork?

Is there a load order that avoids having to compile a special babashka binary with specter?

I undersand specter_sci  to be a special use case of specter on the CLI.

What Am I missing?

Thanks for any helpful hints.

**Comment by borkdude** [comment]
Babashka currently doesn't include specter by default, but we could make a feature flag for it so you can compile it yourself with specter. If we would include specter, I'd first like to do a community poll about that like we've done with other libraries like clojure.spec.alpha and malli.

**Comment by borkdude** [comment]
More info on feature flags here: https://github.com/babashka/babashka/blob/master/doc/build.md#feature-flags

**Comment by borkdude** [comment]
If it helps, this tool uses SCI + specter too: https://github.com/markus-wa/cq
